dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx

```diff
@@ -0,0 +1,1701 @@
+                    {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+                    <label>Repository Authorization</label>
+                    <CdsRadio>
+                      <label htmlFor="kubeapps-repo-auth-method-none">None (Public)</label>
+                      <input
+                        id="kubeapps-repo-auth-method-none"
+                        type="radio"
+                        name="auth"
+                        value={
+                          PackageRepositoryAuth_PackageRepositoryAuthType[
+                            PackageRepositoryAuth_PackageRepositoryAuthType
+                              .PACKAGE_REPOSITORY_AUTH_TYPE_UNSPECIFIED
+                          ]
+                        }
+                        checked={
+                          authMethod ===
+                          PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_UNSPECIFIED
+                        }
+                        onChange={handleAuthRadioButtonChange}
+                        disabled={!!repo.auth?.type}
+                      />
+                    </CdsRadio>
+                    <CdsRadio>
+                      <label htmlFor="kubeapps-repo-auth-method-basic">Basic Auth</label>
+                      <input
+                        id="kubeapps-repo-auth-method-basic"
+                        type="radio"
+                        name="auth"
+                        value={
+                          PackageRepositoryAuth_PackageRepositoryAuthType[
+                            PackageRepositoryAuth_PackageRepositoryAuthType
+                              .PACKAGE_REPOSITORY_AUTH_TYPE_BASIC_AUTH
+                          ]
+                        }
+                        checked={
+                          authMethod ===
+                          PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_BASIC_AUTH
+                        }
+                        onChange={handleAuthRadioButtonChange}
+                        disabled={!!repo.auth?.type}
+                      />
+                    </CdsRadio>
+                    <CdsRadio>
+                      <label htmlFor="kubeapps-repo-auth-method-bearer">Bearer Token</label>
+                      <input
+                        id="kubeapps-repo-auth-method-bearer"
+                        type="radio"
+                        name="auth"
+                        value={
+                          PackageRepositoryAuth_PackageRepositoryAuthType[
+                            PackageRepositoryAuth_PackageRepositoryAuthType
+                              .PACKAGE_REPOSITORY_AUTH_TYPE_BEARER
+                          ]
+                        }
+                        checked={
+                          authMethod ===
+                          PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_BEARER
```

Comment:
Not all backend implementations support all auth types defined in the API.
For example, Flux [does not implement some](https://github.com/vmware-tanzu/kubeapps/blob/main/cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go#L1124), or Helm does not implement the SSH for GIt like Carvel.
Maybe we should add some conditions for that in the frontend? Otherwise the user will get an `Unimplemented` error.
This is one of the things that the Plugins customizable UI feature should solve in the future.

Reply:
Good idea! I'll add to this PR tomorrow! 

> This is one of the things that the Plugins customizable UI feature should solve in the future.

Big +1 to this

Reply:
Cool!